File: Battleship.Server/Models/Game.cs
using System;
using System.Collections.Generic;
using System.Dynamic;

namespace Battleship.Server.Models
{
    public class Game
    {
        public Game()
        {
            Players = new List<Player>();
        }

        public long Id { get; set; }

        /// <summary>
        /// Identifier from the url
        /// </summary>
        public string UrlId { get; set; }

        /// <summary>
        /// Players in the game
        /// </summary>
        public IList<Player> Players { get; set; }

        public void AddPlayer(Player player)
        {
            Players.Add(player);
        }

        public static Game Create()
        {
            var game = new Game
            {
                UrlId = Guid.NewGuid().ToString("N")
            };

            return game;
        }
    }
}

File: Battleship.Client/src/models/game.ts
import { User } from './user'
import { GameDto } from './Dto/game-dto';

export class Game{
    players: User[] = [];
    token: string | undefined;

    public get url(): string{
        return "?gid=" + this.token;
    }

    public static create(game: GameDto): Game{
        const result = new Game();
        const players = game.players.map(player => User.create(player));
        result.players = players;
        result.token = game.token;

        return result;
    }
}